cocos/2d/CCParticleSystemQuad.cpp

```diff
@@ -239,11 +239,7 @@ void ParticleSystemQuad::setDisplayFrame(SpriteFrame *spriteFrame)
     CCASSERT(spriteFrame->getOffsetInPixels().isZero(), 
              "QuadParticle only supports SpriteFrames with no offsets");
 
-    // update texture before updating texture rect
-    if ( !_texture || spriteFrame->getTexture()->getName() != _texture->getName())
```

Comment:
Yep, it is ok.
Then could you tell me what's the issue? You are appreciated if you can provide a test case to reproduce it.